teuthology/orchestra/run.py

```diff
@@ -221,6 +237,70 @@ def __repr__(self):
             name=self.hostname,
             )
 
+class ErrorScanner(object):
+    """
+    Scan for unit tests errors in teuthology.log 
+    """
+    _flag = 0
+    ERROR_PATTERNS = {
+        "prev_detected": [re.compile(r"UnitTestError")],
+        "nose": [
+            re.compile(r"ERROR:\s"),
+            re.compile(r"FAIL:\s"),
+        ],
+        "gtest":  [re.compile(r"\[\s\sFAILED\s\s\]")],
+    }
+    
+    def _search_error(self, line, test):
+        test_patterns = ErrorScanner.ERROR_PATTERNS[test]
+
+        for pattern in test_patterns:
+            error = pattern.search(line)
+            if error:
+                is_old_error = self._is_prev_detected_error(line)
+                if not is_old_error:
+                    return line[error.start():].strip()
+        return None
+
+    def _is_prev_detected_error(self, line) -> bool:
```

Comment:
Just to make sure I understood this correctly, you are using prev_detected errors to filter out all the UnitTestErrors in the log file?

Reply:
Yes, because UnitTestErrors exceptions have the error message - which includes the error pattern (like "ERROR:") - so this is to ensure we avoid reading these exceptions as original failures.

Reply:
Could you please give an example why this is needed?